<a href="https://colab.research.google.com/github/Hamzaishappy786/Side-projects/blob/main/Generative%20AI/2.%20Article%20summarizer%20(Made%20with%20Gemini%20API)/2.%20Article_summarizer_(Made_with_Gemini_API).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Simple Document summarizer(Made with Gemini API)

## Dependencies to install:
1. PyPDF2
2. docx
3. google.generativeai
4. exceptions (just in case)

In [3]:
!pip install PyPDF2 python-docx

  Using cached pypdf2-3.0.1-py3-none-any.whl.metadata (6.8 kB)
  Using cached python_docx-1.1.2-py3-none-any.whl.metadata (2.0 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 232.6/232.6 kB 7.2 MB/s eta 0:00:00
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 244.3/244.3 kB 20.3 MB/s eta 0:00:00


### Add a file in your colab, to paste file to.

In [5]:
import google.generativeai as genai
import PyPDF2
import docx
import os

# 🔹 Configure Gemini API (Replace with your API key)
genai.configure(api_key="AIzaSyBsfG_yYl8NnCG34gspFZXpgWpzPOCTdg8")

# 🔹 Initialize the model
model = genai.GenerativeModel("gemini-1.5-pro")

# 🔹 Function to extract text from PDF
def extract_text_from_pdf(file_path):
    text = ""
    try:
        with open(file_path, "rb") as file:
            reader = PyPDF2.PdfReader(file)
            for page in reader.pages:
                extracted_text = page.extract_text()
                if extracted_text:
                    text += extracted_text + "\n"
    except Exception as e:
        print(f"⚠️ Error processing PDF: {e}")
    return text.strip() if text else "⚠️ Could not extract text from PDF."

# 🔹 Function to extract text from DOCX
def extract_text_from_docx(file_path):
    text = ""
    try:
        doc = docx.Document(file_path)
        for para in doc.paragraphs:
            text += para.text + "\n"
    except Exception as e:
        print(f"⚠️ Error processing DOCX: {e}")
    return text.strip() if text else "⚠️ Could not extract text from DOCX."

# 🔹 Function to summarize text using Gemini API
def summarize_text(text):
    prompt = f"Summarize the following text in 3-4 sentences. Make sure that the crux in the text that you got stays easy and nuetral for users to read. Provide users with points as your lines:\n\n{text}"
    response = model.generate_content(prompt)
    return response.text if response.text else "⚠️ No summary generated."

# 🔹 Main function (No Tkinter)
def main():
    file_path = input("📄 Enter the file path (PDF or DOCX): ").strip()

    # Validate file existence
    if not os.path.exists(file_path):
        print("⚠️ File not found. Please check the path.")
        return

    # Detect file type and extract text
    if file_path.lower().endswith(".pdf"):
        text = extract_text_from_pdf(file_path)
    elif file_path.lower().endswith(".docx"):
        text = extract_text_from_docx(file_path)
    else:
        print("⚠️ Unsupported file format. Use PDF or DOCX.")
        return

    # Summarize the extracted text
    if text:
        summary = summarize_text(text)
        print("\n📌 **Summary:**\n", summary)
    else:
        print("⚠️ No text extracted from the document.")

# Run the program
if __name__ == "__main__": main()

📄 Enter the file path (PDF or DOCX): /content/Writing Instructions in Technical Writing.pdf

📌 **Summary:**
 * Instructions are frequently written by technical communicators, but other professionals also create them for various documents.  Clear instructions are crucial for user safety and product usability.
* Effective instructions consider the audience's background and skill level, using appropriate vocabulary.  Because readers approach instructions differently, writers should assume a step-by-step reading style and organize instructions chronologically.
* Key elements of instructions include a specific title, a clear purpose (if not obvious), listed materials/tools/conditions, numbered steps in present tense/imperative mode, and a conclusion explaining the outcome.
* Instructions should be organized logically with clear visuals, separating supplemental information. Safety information should be prominently displayed and scaled in emphasis based on the level of risk (danger, warning, 

<div style="
    font-family: 'Arial', sans-serif;
    font-size: 36px;
    font-weight: bold;
    color: white;
    background: linear-gradient(45deg, #ff416c, #ff4b2b);
    padding: 20px;
    text-align: center;
    border-radius: 15px;
    box-shadow: 0px 4px 10px rgba(0, 0, 0, 0.2);
    animation: fadeOut 3s ease-in-out infinite alternate;">
    🚀 Have a nice day. Good bye! 👋
</div>

<style>
@keyframes fadeOut {
    0% { opacity: 1; transform: scale(1); }
    100% { opacity: 0.6; transform: scale(1.1); }
}
</style>
